db/internal_stats.cc

```diff
@@ -580,6 +584,9 @@ const UnorderedMap<std::string, DBPropertyInfo>
         {DB::Properties::kNumRunningCompactions,
          {false, nullptr, &InternalStats::HandleNumRunningCompactions, nullptr,
           nullptr}},
+        {DB::Properties::kNumRunningCompactionSortedRuns,
+         {false, nullptr, &InternalStats::HandleNumRunningCompactionSortedRuns,
```

Comment:
I believe the first parameter "false" indicates we should call the handler function under db lock. Therefore lock is no need to be acquired inside HandleNumRunningCompactionSortedRuns()

Reply:
> I believe the first parameter "false" indicates we should call the handler function under db lock

It seems like that, though the naming of `need_out_of_mutex` is confusing

For instance I found this:
```cpp
  } else if (property_info->handle_string) {
    if (property_info->need_out_of_mutex) {
      return cfd->internal_stats()->GetStringProperty(*property_info, property,
                                                      value);
    } else {
      InstrumentedMutexLock l(&mutex_);
      return cfd->internal_stats()->GetStringProperty(*property_info, property,
                                                      value);
    }
  } 
```

Reply:
hmm - which part of that is confusing? i believe need_out_of_mutex true indicated we don't need `InstrumentedMutexLock l(&mutex_);`

Reply:
I see - there is some room for improving the naming then